cranelift/codegen/src/ir/sourceloc.rs

```diff
@@ -51,6 +51,61 @@ impl fmt::Display for SourceLoc {
     }
 }
 
+/// Source location relative to another base source location.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+pub struct RelSourceLoc(u32);
```

Comment:
Does using RelSourceLoc the same way as SourceLoc give correct results? In cg_clif SourceLoc is already function local. (It is backed by an IndexSet with mir::SourceLoc as entries)

Reply:
Yes, it's mostly used internally so I wouldn't even expect embedders to interact with it.